zerver/lib/alert_words.py

```diff
@@ -1,18 +1,16 @@
-from django.db.models import Q
-from zerver.models import UserProfile, Realm
+from zerver.models import UserProfile, Realm, AlertWord
 from zerver.lib.cache import cache_with_key, realm_alert_words_cache_key, \
     realm_alert_words_automaton_cache_key
-import ujson
 import ahocorasick
 from typing import Dict, Iterable, List
 
 @cache_with_key(realm_alert_words_cache_key, timeout=3600*24)
 def alert_words_in_realm(realm: Realm) -> Dict[int, List[str]]:
-    users_query = UserProfile.objects.filter(realm=realm, is_active=True)
-    alert_word_data = users_query.filter(~Q(alert_words=ujson.dumps([]))).values('id', 'alert_words')
-    all_user_words = dict((elt['id'], ujson.loads(elt['alert_words'])) for elt in alert_word_data)
-    user_ids_with_words = dict((user_id, w) for (user_id, w) in all_user_words.items() if len(w))
-    return user_ids_with_words
+    queries = AlertWord.objects.filter(realm=realm).values("user_profile", "word")
```

Comment:
`.values("user_profile_id")` is clearer that we're just getting the ID, not a whole UserProfile object, which would be a waste given what follows.